### PR TITLE
[Inductor] [Quant] Enable lowering of quant per tensor and refactor quant pattern

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #125207
* #124246
* __->__ #124041

**Summary**
Per the discussion in https://github.com/pytorch/pytorch/pull/123444, the `decomposed quant/dequant` patterns changed after https://github.com/pytorch/pytorch/pull/123445, we can move the optimization of `decomposed quant/dequant` from inductor decomposition into lowering phase to avoid the changes. In this way, we can:

- Avoid the pattern matcher failure introduced in https://github.com/pytorch/pytorch/pull/123445
- Make the quantization pattern clearer in the pattern matcher phase, since the `quant/dequant` nodes have not been decomposed.

**Changes in this PR**

- Move optimization of `decomposed quant/dequant` from inductor decomposition into lowering phase.
- Corresponding changes in the quantization pattern matcher to ensure no bc-breaking.

**TestPlan**
```
python -u -m pytest -s -v test/inductor/test_mkldnn_pattern_matcher.py -k test_q
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
Hi @peterbell10 @jerryzh168 @jgong5, per the discussion in https://github.com/pytorch/pytorch/pull/123444#discussion_r1561873118, I have draft the implementation based on option 1 as moving optimization of decomposed quant/dequant from inductor decomposition into lowering phase. Please kindly help to take a look.

File: aten/src/ATen/native/quantized/cpu/qconv.cpp [hunk 1]
```diff
@@ -1659,7 +1659,7 @@ static at::Tensor _quantized_convolution_onednn(
   ideep::convolution_forward::prepare(
       params, src, packed_weight, expected_bias, dst_dims, dst,
       stride.vec(), dilation.vec(), padding.vec(), padding.vec(), groups,
-      src_scales, weights_scales, ideep::scale_t(1, inv_output_scale),
+      src_scales, weights_scales, ideep::scale_t(1, 1.0f / inv_output_scale),
```
Comment (jgong5) [inline aten/src/ATen/native/quantized/cpu/qconv.cpp:1662]: Is it a bug fix or what?

Comment (leslie-fang-intel) [reply aten/src/ATen/native/quantized/cpu/qconv.cpp:1662]: No, it's changed as expected. Previously, we do the quant decomposition, there is a node as `inv_scale = 1 / scale` from decomposition which will be constant folded. So, we get the `inv_scale` actually.

Comment (jgong5) [reply aten/src/ATen/native/quantized/cpu/qconv.cpp:1662]: Then the variable name sounds confusing... It is still `inv_output_scale` or `output_scale`?

Comment (leslie-fang-intel) [reply aten/src/ATen/native/quantized/cpu/qconv.cpp:1662]: Stack PR: https://github.com/pytorch/pytorch/pull/124246 to change the name to `output_scale`.

**Review by peterbell10 — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

**Comment by leslie-fang-intel** [comment]
> @pytorchbot revert -m 'Sorry for reverting your change but I think there is a land race with the change https://hud.pytorch.org/pytorch/pytorch/commit/33e6791645b5950b0f39301f55b8a4a79c0ca847' -c landrace

Hi @huydhn, has this change https://hud.pytorch.org/pytorch/pytorch/commit/33e6791645b5950b0f39301f55b8a4a79c0ca847' landed? Should I rebase this ghstack and try land again?

**Comment by huydhn** [comment]
Yes, please do a rebase to main and try to land this again

**Comment by huydhn** [comment]
The stack is probably having a landrace with this change https://github.com/pytorch/pytorch/pull/122832 that was landed few hours ago

**Comment by leslie-fang-intel** [comment]
> The stack is probably having a landrace with this change #122832 that was landed few hours ago

Rebase to main and check the preCI again.

**Comment by leslie-fang-intel** [comment]
> Yes, please do a rebase to main and try to land this again

Hi @huydhn, after rebase, the preCI are all green. I am going to re-land.

**Pull request closed without merging**